src/Factory/CreditMemoFactory.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\RefundPlugin\Factory;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\RefundPlugin\Entity\CreditMemoInterface;
+use Sylius\RefundPlugin\Entity\CustomerBillingDataInterface;
+use Sylius\RefundPlugin\Entity\ShopBillingDataInterface;
+use Sylius\RefundPlugin\Generator\CreditMemoIdentifierGeneratorInterface;
+use Sylius\RefundPlugin\Generator\NumberGenerator;
+use Sylius\RefundPlugin\Provider\CurrentDateTimeImmutableProviderInterface;
+use Webmozart\Assert\Assert;
+
+final class CreditMemoFactory implements CreditMemoFactoryInterface
+{
+    /** @var FactoryInterface */
+    private $creditMemoFactory;
+
+    /** @var CreditMemoIdentifierGeneratorInterface */
+    private $creditMemoIdentifierGenerator;
+
+    /** @var NumberGenerator */
+    private $creditMemoNumberGenerator;
+
+    /** @var CurrentDateTimeImmutableProviderInterface */
+    private $currentDateTimeImmutableProvider;
+
+    public function __construct(
+        FactoryInterface $creditMemoFactory,
+        CreditMemoIdentifierGeneratorInterface $creditMemoIdentifierGenerator,
+        NumberGenerator $creditMemoNumberGenerator,
+        CurrentDateTimeImmutableProviderInterface $currentDateTimeImmutableProvider
+    ) {
+        $this->creditMemoFactory = $creditMemoFactory;
+        $this->creditMemoIdentifierGenerator = $creditMemoIdentifierGenerator;
+        $this->creditMemoNumberGenerator = $creditMemoNumberGenerator;
+        $this->currentDateTimeImmutableProvider = $currentDateTimeImmutableProvider;
+    }
+
+    public function createNew(): CreditMemoInterface
+    {
+        /** @var CreditMemoInterface $creditMemo */
+        $creditMemo = $this->creditMemoFactory->createNew();
+
+        return $creditMemo;
```

Comment:
```suggestion
        $return $this->creditMemoFactory->createNew();
```

Maybe this will work?

Reply:
I need to add a type hint to this variable because of static analysis